### PR TITLE
Switch CI to use GitHub Actions

### PR DESCRIPTION
**Pull request by fingolfin**

### CODE REVIEW

**Comment by wilfwilson** [comment]
I have re-run the tests after updating the `gap-actions/compile-documentation-for-packages` action, and the 'Build manuals' job still fails, but I believe that is not the fault of the action.

I also get the same failure on my local machine. I believe it is caused because GAP no longer (as of `stable-4.11`) contains the file `doc/mrabbrev.bib`.

**Comment by fingolfin** [comment]
Thanks. But weird, I still can't reproduce this locally, despite `mrabbrev.bib` being gone.

And good that you mention cubefree: because it actually passed that test yesterday, see https://github.com/gap-packages/cubefree/actions. Did perhaps something else change in the Github Action?

**Comment by wilfwilson** [comment]
The GitHub action passed yesterday because my PR for dealing with `doc/make_doc` hadn't been merged yet. The previous behaviour (as exhibited in this run https://github.com/gap-packages/cubefree/runs/2273374598) was as follows:

* Load GAP and read the file `makedoc.g`
* The file does not exist, so GAP quits and doesn't do anything
* GitHub Actions tries to upload the artifacts, but since none exist, it doesn't actually do anything (there is a warning about that visible https://github.com/gap-packages/cubefree/actions/runs/720568938).

But despite not doing anything, no failure was caused - which is why it passed yesterday.

**Comment by wilfwilson** [comment]
To make this more robust in the future, the 'upload' action should probably be made to fail the job if no artifact is produced.

**Comment by fingolfin** [comment]
Ok thanks! Now I just habe to figure out why it doesn't fail for me locally. Well, that and a fix

**Comment by wilfwilson** [comment]
What exit code do you get after running `./make_doc` for ace/cubefree? Although, for me locally, the `make_doc` script produces a reasonable-looking PDF for cubefree, and a reasonable-looking DVI for ace (although no PDF), in both cases it gives an exit code of `2`.

**Comment by wilfwilson** [comment]
(And if I put the `mrabbrev.bib` file back in `<GAPROOT>/doc`, `ace` produces a PDF, and both packages return `0` after running `make_doc`)

**Comment by wilfwilson** [comment]
Unfortunately it now fails because it tries to include the GAP file `manualbib.xml.bib`, which I gather only is created when compiling the GAP manuals. It uses this file for three bibitems (which already are almost exactly duplicated in ace's `doc/manual.bib`, albeit commented out).

I see a couple of ways to proceed:
* `gap-actions/compile-documentation-for-packages` could have an option that compiles the GAP manuals before compiling the package manuals (ultimately, we could provide docker images with pre-compiled manuals).
* Remove the inclusion of `manualbib.xml.bib` in `ace`, and instead reinstate the three bibitems in ace's `manual.bib`.

**Comment by fingolfin** [comment]
I'll remove the use of `manualbib.xml.bib`

File: doc/manual.bib [hunk 1]
```diff
@@ -128,19 +128,16 @@ @INCOLLECTION{Bee84
...
-%%   AUTHOR        = {George Havas},
-%%   TITLE         = {Coset enumeration strategies},
-%%   BOOKTITLE     = {ISSAC'91 (Proceedings of the 1991 International
-%%                         Symposium on Symbolic and Algebraic Computation)},
-%%   YEAR          = {1991},
-%%   EDITOR        = {Stephen M. Watt},
-%%   PAGES         = {191--199},
-%%   PUBLISHER     = {ACM Press}
-%% }
+INPROCEEDINGS{Hav91,
```
Comment (wilfwilson) [inline doc/manual.bib:131]: These items need the `@` adding:
```suggestion
@INPROCEEDINGS{Hav91,
```

**Review by wilfwilson — COMMENTED** [review]
(Some of) the uncommented bibitems in this the `doc/manual.bib` file differ from the equivalent items in GAP's `doc/manualbib.xml.bib` - should we worry about this? (Only in terms of the amount of detail/formatting, not in terms of `citekey`s or what papers they're referring to).

**Review by wilfwilson — APPROVED** [review]
I pushed the fix to the bib file that I suggested. There is still this question, but I'm minded to say it's not a problem:

> (Some of) the uncommented bibitems in this the `doc/manual.bib` file differ from the equivalent items in GAP's `doc/manualbib.xml.bib` - should we worry about this? (Only in terms of the amount of detail/formatting, not in terms of `citekey`s or what papers they're referring to).

**Pull request merged**